lib/Frontend/ParseableInterfaceSupport.cpp

```diff
@@ -0,0 +1,325 @@
+      return InStatus.get().getLastModificationTime() <=
+        OutStatus.get().getLastModificationTime();
+    }
+  }
+  return false;
+}
+
+static bool buildSwiftModuleFromSwiftInterface(
+    clang::vfs::FileSystem &FS, DiagnosticEngine &Diags,
+    CompilerInvocation &SubInvocation, StringRef InPath, StringRef OutPath) {
+  bool SubError = false;
+  bool RunSuccess = llvm::CrashRecoveryContext().RunSafelyOnThread([&] {
+
+    llvm::BumpPtrAllocator SubArgsAlloc;
+    llvm::StringSaver SubArgSaver(SubArgsAlloc);
+    SmallVector<const char *, 16> SubArgs;
+    swift::version::Version Vers;
+    if (extractSwiftInterfaceVersionAndArgs(Diags, FS, InPath, Vers,
+                                            SubArgSaver, SubArgs)) {
+      SubError = true;
+      return;
+    }
+
+    if (SubInvocation.parseArgs(SubArgs, Diags)) {
+      SubError = true;
+      return;
+    }
+
+    // Build the .swiftmodule; this is a _very_ abridged version of the logic in
+    // performCompile in libFrontendTool, specialized, to just the one
+    // module-serialization task we're trying to do here.
+    LLVM_DEBUG(llvm::dbgs() << "Setting up instance\n");
+    CompilerInstance SubInstance;
+    if (SubInstance.setup(SubInvocation)) {
+      SubError = true;
+      return;
+    }
+
+    LLVM_DEBUG(llvm::dbgs() << "Performing sema\n");
+    SubInstance.performSema();
+    if (SubInstance.getASTContext().hadError()) {
+      SubError = true;
+      return;
+    }
+
+    auto Mod = SubInstance.getMainModule();
+    auto SILMod = SubInstance.takeSILModule();
+    if (SILMod) {
+      LLVM_DEBUG(llvm::dbgs() << "Running SIL diagnostic passes\n");
+      if (runSILDiagnosticPasses(*SILMod)) {
+        SubError = true;
+        return;
+      }
+    }
+
+    LLVM_DEBUG(llvm::dbgs() << "Serializing " << OutPath << "\n");
+    SerializationOptions serializationOpts;
+    std::string OutPathStr = OutPath;
+    serializationOpts.OutputPath = OutPathStr.c_str();
+    serializationOpts.SerializeAllSIL = true;
+    serialize(Mod, serializationOpts, SILMod.get());
+    SubError = Diags.hadAnyError();
+  });
+  return !RunSuccess || SubError;
+}
+
+/// Load a .swiftmodule associated with a .swiftinterface either from a
+/// cache or by converting it in a subordinate \c CompilerInstance, caching
+/// the results.
+std::error_code ParseableInterfaceModuleLoader::openModuleFiles(
+    StringRef DirName, StringRef ModuleFilename, StringRef ModuleDocFilename,
+    std::unique_ptr<llvm::MemoryBuffer> *ModuleBuffer,
+    std::unique_ptr<llvm::MemoryBuffer> *ModuleDocBuffer,
+    llvm::SmallVectorImpl<char> &Scratch) {
+
+  auto &FS = *Ctx.SourceMgr.getFileSystem();
+  auto &Diags = Ctx.Diags;
+  llvm::SmallString<128> InPath, OutPath;
+
+  // First check to see if the .swiftinterface exists at all. Bail if not.
+  InPath = DirName;
+  llvm::sys::path::append(InPath, ModuleFilename);
+  auto Ext = file_types::getExtension(file_types::TY_SwiftModuleInterfaceFile);
```

Comment:
Oh, we maybe should have changed `TY_SwiftModuleInterfaceFile` to `TY_SwiftParseableInterfaceFile`

Reply:
done